From 78dae329abff0271b425afe9a2104c6ecea1499d Mon Sep 17 00:00:00 2001
From: Cameron Clark <cameron.clark@hey.com>
Date: Fri, 14 Nov 2025 11:09:52 +0000
Subject: Adapt project service for single run mode

---
 internal/project/project.go | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/internal/project/project.go b/internal/project/project.go
index b54b8b1b7..b8526fea9 100644
--- a/internal/project/project.go
+++ b/internal/project/project.go
@@ -2,6 +2,7 @@ package project
 
 import (
 	"fmt"
+	"runtime"
 	"strings"
 	"sync"
 
@@ -346,7 +347,7 @@ func (p *Project) CreateProgram() CreateProgramResult {
 				TypingsLocation:             typingsLocation,
 				JSDocParsingMode:            ast.JSDocParsingModeParseAll,
 				CreateCheckerPool: func(program *compiler.Program) compiler.CheckerPool {
-					checkerPool = newCheckerPool(4, program, p.log)
+					checkerPool = newCheckerPool(runtime.GOMAXPROCS(0), program, p.log)
 					return checkerPool
 				},
 			},
-- 
2.50.1 (Apple Git-155)

