From d0efefdc1fc4a9293e0a7b80c32afb673407e0fc Mon Sep 17 00:00:00 2001
From: Cameron Clark <cameron.clark@hey.com>
Date: Fri, 14 Nov 2025 11:09:52 +0000
Subject: [PATCH 2/5] Adapt project service for single run mode

---
 internal/project/project.go | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/internal/project/project.go b/internal/project/project.go
index 5b9e2fd8eb..3e83c51a64 100644
--- a/internal/project/project.go
+++ b/internal/project/project.go
@@ -2,6 +2,7 @@ package project
 
 import (
 	"fmt"
+	"runtime"
 	"strings"
 	"sync"
 
@@ -343,7 +344,7 @@ func (p *Project) CreateProgram() CreateProgramResult {
 				TypingsLocation:             typingsLocation,
 				JSDocParsingMode:            ast.JSDocParsingModeParseAll,
 				CreateCheckerPool: func(program *compiler.Program) compiler.CheckerPool {
-					checkerPool = newCheckerPool(4, program, p.log)
+					checkerPool = newCheckerPool(runtime.GOMAXPROCS(0), program, p.log)
 					return checkerPool
 				},
 			},
-- 
2.50.1 (Apple Git-155)

