From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cameron Clark <cameron.clark@hey.com>
Date: Fri, 14 Nov 2025 11:09:52 +0000
Subject: [PATCH 2/6] Adapt project service for single run mode


diff --git a/internal/project/project.go b/internal/project/project.go
index 330509065cdc0db9ecb7f8823779d20db11ef996..46c1a790cf1a361020f6e2a694b262075adea538 100644
--- a/internal/project/project.go
+++ b/internal/project/project.go
@@ -2,6 +2,7 @@ package project
 
 import (
 	"fmt"
+	"runtime"
 	"strings"
 	"sync"
 
@@ -326,7 +327,7 @@ func (p *Project) CreateProgram() CreateProgramResult {
 				UseSourceOfProjectReference: true,
 				TypingsLocation:             typingsLocation,
 				CreateCheckerPool: func(program *compiler.Program) compiler.CheckerPool {
-					checkerPool = newCheckerPool(4, program, p.log)
+					checkerPool = newCheckerPool(runtime.GOMAXPROCS(0), program, p.log)
 					return checkerPool
 				},
 			},
