From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cameron Clark <cameron.clark@hey.com>
Date: Mon, 3 Nov 2025 12:44:32 +0000
Subject: [PATCH 4/6] fix: early return from invalid tsconfig for better errors


diff --git a/internal/tsoptions/tsconfigparsing.go b/internal/tsoptions/tsconfigparsing.go
index 2392d9d371850829d019dd999d7335b178f69de9..05b7af52f5f84abadc8868b56df3ec6bd91e7837 100644
--- a/internal/tsoptions/tsconfigparsing.go
+++ b/internal/tsoptions/tsconfigparsing.go
@@ -192,6 +192,17 @@ func parseOwnConfigOfJsonSourceFile(
 	var extendedConfigPath any
 	var rootCompilerOptions []*ast.PropertyName
 	var errors []*ast.Diagnostic
+
+	// If there are parsing errors, return only the first one
+	if len(sourceFile.Diagnostics()) > 0 {
+		return &parsedTsconfig{
+			raw:                &collections.OrderedMap[string, any]{},
+			options:            compilerOptions,
+			typeAcquisition:    typeAcquisition,
+			extendedConfigPath: extendedConfigPath,
+		}, []*ast.Diagnostic{sourceFile.Diagnostics()[0]}
+	}
+
 	onPropertySet := func(
 		keyText string,
 		value any,
