From 5f2b70a9908160df84f6b37c8480c8f94aad117d Mon Sep 17 00:00:00 2001
From: Noel Kim <react-native@naver.com>
Date: Thu, 11 Sep 2025 11:46:56 +0900
Subject: [PATCH 5/5] feat(checker/types): add public getter from `TypePredicate`

---
 internal/checker/types.go | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/internal/checker/types.go b/internal/checker/types.go
index 5fd4b1583..c8bdf3e98 100644
--- a/internal/checker/types.go
+++ b/internal/checker/types.go
@@ -1192,6 +1192,14 @@ type TypePredicate struct {
 	t              *Type
 }
 
+func (t *TypePredicate) Kind() TypePredicateKind { return t.kind }
+
+func (t *TypePredicate) ParameterIndex() int32 { return t.parameterIndex }
+
+func (t *TypePredicate) ParameterName() string { return t.parameterName }
+
+func (t *TypePredicate) Type() *Type { return t.t }
+
 // IndexInfo
 
 type IndexInfo struct {
-- 
2.39.5 (Apple Git-154)

